### PR TITLE
Add search bar and filter by date for real events in home page

### PR DESCRIPTION
**Pull request by gabrielorleans**
image

Search bar now functional and displaying only (1) events published and (2) yet to happen

### CODE REVIEW

File: yarn.lock
Comment (luizakp) [inline yarn.lock:1]: 👀

**Review by luizakp — CHANGES_REQUESTED** [review]
This is not getting all events

File: src/app/page.tsx [hunk 1]
```diff
@@ -53,10 +51,10 @@ function sortByStartsAt(data: DataEntry[]): DataEntry[] {
...
 export default function Events() {
 	const signer = useSigner();
-
 	const chainId = useChainId();
-
 	const attester = signer?.address || "";
+	const [searchTerm, setSearchTerm] = useState("");
+
 	const [result] = useQuery({
 		query: USER_ATTESTATIONS_QUERY,
```
Comment (luizakp) [inline src/app/page.tsx:59]: this is the wrong Query... we talked about it on the daily. You should get all the events, so all the attestations that has the same schema ID

File: src/app/page.tsx [hunk 2]
```diff
@@ -113,15 +152,15 @@ function AttestationItem({ data }: { data: any }) {
...
+			className="border-2 rounded-lg w-[80vw] sm:w-[70vw] md:w-[50vw] lg:w-[25vw] h-auto gap-6 justify-self-center"
+			>
 			<div className="p-3">
 				<h2 className="text-base font-bold">{parsedData.name}</h2>
 				<p className="text-xs">{epochToCustomDate(parsedData.startsAt)}</p>
 			</div>
 			<div>
 				<img
-					className="w-full h-auto"
+					className="w-full h-32"
```
Comment (luizakp) [inline src/app/page.tsx:163]: why this card is different from the one on events page?

Comment (luizakp) [reply src/app/page.tsx:163]: Home Page one:

```
function AttestationItem({ data }: { data: any }) {
  const parsedData = parseEventsData(data);

  return (
    <div
      key="Event"
      className="border-2 rounded-lg w-[80vw] sm:w-[70vw] md:w-[50vw] lg:w-[25vw] h-auto gap-6 justify-self-center"
    >
      <div className="p-3">
        <h2 className="text-base font-bold">{parsedData.name}</h2>
        <p className="text-xs">{epochToCustomDate(parsedData.startsAt)}</p>
      </div>
      <div>
        <img
          className="w-full h-32"
          alt="Template image"
          src={parsedData.imageUrl}
        />
      </div>
      <div className="p-3">
        <p className="text-xs text-gray-700 mb-4">
          Brief description: {parsedData.briefDescription}
        </p>
        <span className="text-xs bg-gray-100 px-2 py-1 rounded-lg inline-block">
          Status: published
        </span>
      </div>
    </div>
  );
}
```

Event page one:

```
function AttestationItem({ id, data }: { id: string; data: any }) {
  const parsedData = parseEventsData(data);

  return (
    <Link
      href={`/event/${id}`}
      className="border-2 rounded-lg w-60 hover:border-slate-600"
    >
      <div className="p-3">
        <h1 className="text-base font-bold">{parsedData.name}</h1>
        <h2 className="text-xs">{parsedData.startsAt}</h2>
      </div>
      <div>
        <img
          className="w-full h-auto"
          alt="Template image"
          src={parsedData.imageUrl}
        />
      </div>
      <div className="p-3">
        <h2 className="text-xs text-gray-700 mb-4">
          Brief description: {parsedData.briefDescription}
        </h2>
        <h2 className="text-xs text-gray-700 mb-4">
          Full description: {parsedData.fullDescription}
        </h2>
      </div>
    </Link>
  );
}
```

File: src/app/page.tsx [hunk 3]
```diff
@@ -113,15 +152,15 @@ function AttestationItem({ data }: { data: any }) {
 	return (
 		<div
 			key="Event"
-			className="border-2 rounded-lg w-60 h-45 gap-6 justify-self-center"
-		>
+			className="border-2 rounded-lg w-[80vw] sm:w-[70vw] md:w-[50vw] lg:w-[25vw] h-auto gap-6 justify-self-center"
+			>
```
Comment (luizakp) [inline src/app/page.tsx:156]: this is not linked to the event details page 🤔

File: src/app/page.tsx [hunk 4]
```diff
@@ -66,43 +64,84 @@ export default function Events() {
...
+		const currentTime = Date.now();
+	  
+		return sortByStartsAt(data.attestations).filter((attestation) => {
+		  const decodedData = JSON.parse(attestation.decodedDataJson);
+		  const parsedData = parseEventsData(attestation.decodedDataJson);
+		  
+		  // Filter out past events
+		  if (parsedData.startsAt <= currentTime) {
+			return false;
+		  }
```
Comment (luizakp) [inline src/app/page.tsx:78]: startsAt can be in the past, endsAt that can't

File: src/app/page.tsx [hunk 5]
```diff
@@ -4,24 +4,22 @@ import { USER_ATTESTATIONS_QUERY } from "@/lib/gqlEasAttestation/query";
...
-import { useMemo } from "react";
 import { parseEventsData } from "./events/parseEventsData";
-import { fromUnixTime, format } from 'date-fns';
-import { enUS } from 'date-fns/locale';
-
+import { fromUnixTime, format } from "date-fns";
+import { enUS } from "date-fns/locale";
+import React, { useState, useMemo } from "react";
 
 function epochToCustomDate(epoch: number): string {
```
Comment (luizakp) [inline src/app/page.tsx:12]: this already exists on events/page, no?

**Comment by ribeirojose** [comment]
@JeanNeiverth when you close issues, can you post a comment just briefly explaining why they were closed? It's a bit uncanny to me to come to an issue where my review had been requested to see it closed and not know what happened 😬

**Comment by luizakp** [comment]
Context: This PR was closed because it didn't deliver the task scope. A new one was created:  https://github.com/bleu/op-bacchus/pull/27

**Pull request closed without merging**